ppafm/common.py

```diff
@@ -436,90 +443,86 @@ def maxAlongDirEntropy(atoms, hdir, beta=1.0):
+                        print(key, getattr(parameters, key), words[1])
                 elif isinstance(val, str):
-                    params[key] = words[1]
-                    if verbose > 0:
-                        print(key, params[key], words[1])
-                elif isinstance(val, np.ndarray):
-                    if val.dtype == float:
-                        params[key] = np.array([float(words[1]), float(words[2]), float(words[3])])
+                    setattr(parameters, key, words[1])
+                elif isinstance(val, list):
+                    if isinstance(val[0], float):
+                        setattr(parameters, key, [float(words[1]), float(words[2]), float(words[3])])
                         if verbose > 0:
-                            print(key, params[key], words[1], words[2], words[3])
-                    elif val.dtype == int:
+                            print(key, getattr(parameters, key), words[1], words[2], words[3])
+                    elif isinstance(val[0], int):
                         if verbose > 0:
                             print(key)
-                        params[key] = np.array([int(words[1]), int(words[2]), int(words[3])])
+                        setattr(parameters, key, [int(words[1]), int(words[2]), int(words[3])])
                         if verbose > 0:
-                            print(key, params[key], words[1], words[2], words[3])
+                            print(key, getattr(parameters, key), words[1], words[2], words[3])
                     else:
-                        params[key] = np.array([str(words[1]), float(words[2])])
+                        setattr(parameters, key, [str(words[1]), float(words[2])])
                         if verbose > 0:
-                            print(key, params[key], words[1], words[2])
+                            print(key, getattr(parameters, key), words[1], words[2])
             else:
                 raise ValueError(f"Parameter {key} is not known")
     fin.close()
-    if params["gridN"][0] <= 0:
-        autoGridN()
-
-    params["tip"] = params["tip"].replace('"', "")
-    params["tip"] = params["tip"].replace("'", "")
-    ### necessary for working even with quotemarks in params.ini
-    params["tip_base"][0] = params["tip_base"][0].replace('"', "")
-    params["tip_base"][0] = params["tip_base"][0].replace("'", "")
-    ### necessary for working even with quotemarks in params.ini
+    if parameters.gridN[0] <= 0:
+        autoGridN(parameters=parameters)
+
+    parameters.tip = parameters.tip.replace('"', "")
+    parameters.tip = parameters.tip.replace("'", "")
+    # Necessary for working even with quotemarks in params.ini
+    parameters.tip_base[0] = parameters.tip_base[0].replace('"', "")
+    parameters.tip_base[0] = parameters.tip_base[0].replace("'", "")
+    if verbose > 0:
+        print("Loaded parameters from ", fname)
 
 
-def apply_options(opt):
+def apply_options(opt, parameters):
```

Comment:
Maybe this function as a method of `PpafmParameters` as well?

Reply:
done in https://github.com/Probe-Particle/ppafm/pull/275/commits/1903af182ca1ffe78fffe8e3d7fd3087aca5c6aa